test/badges.test.ts

```diff
@@ -8,129 +8,153 @@ import { splitSignature, _TypedDataEncoder } from 'ethers/lib/utils'
-        const signature = await issuer._signTypedData(typedData.domain, typedData.types, typedData.value)
-        const { compact } = splitSignature(signature)
-        const randomWallet = Wallet.createRandom()
-
-        await expect(
-            badgesContract.connect(claimant).mintWithPermission(randomWallet.address, typedData.value.tokenURI, compact)
-        ).to.be.revertedWith('mintWithPermission: invalid signature')
-    })
-
-    it('should fail to mint when using incorrect token uri', async () => {
-        const { badgesContract, typedData, issuer, claimant } = await loadFixture(deployContractFixture)
-        const signature = await issuer._signTypedData(typedData.domain, typedData.types, typedData.value)
-        const { compact } = splitSignature(signature)
-        await expect(
-            badgesContract
-                .connect(claimant)
-                .mintWithPermission(typedData.value.from, 'https://some-incorrect-uri.com', compact)
-        ).to.be.revertedWith('mintWithPermission: invalid signature')
-    })
-
-    it('should fail to mint when using invalid signature', async () => {
-        const { badgesContract, typedData, issuer, claimant } = await loadFixture(deployContractFixture)
-        const sig = { compact: 'junk conpact value' }
-        const sigAsBytes = ethers.utils.hexlify(ethers.utils.toUtf8Bytes(JSON.stringify(sig)))
-
-        await expect(
-            badgesContract
-                .connect(claimant)
-                .mintWithPermission(typedData.value.from, typedData.value.tokenURI, sigAsBytes)
-        ).to.be.revertedWith('mintWithPermission: invalid signature')
-    })
-
-    it('should fail to mint when using unauthorized claimant', async () => {
-        const { badgesContract, typedData, issuer, claimant, badActor } = await loadFixture(deployContractFixture)
-        const signature = await badActor._signTypedData(typedData.domain, typedData.types, typedData.value)
-        const { compact } = splitSignature(signature)
-
-        await expect(
-            badgesContract.connect(claimant).mintWithPermission(issuer.address, typedData.value.tokenURI, compact)
-        ).to.be.revertedWith('mintWithPermission: invalid signature')
-    })
+    return { badges, badgesContract, owner, issuer, claimant, badActor, typedData }
+  }
+
+  it('should deploy the contract with the right params', async function () {
+    const { badgesContract } = await loadFixture(deployContractFixture)
+
+    const deployedContractName = await badgesContract.name()
+    const deployedSymbolName = await badgesContract.symbol()
+    expect(deployedContractName).to.equal(name)
+    expect(deployedSymbolName).to.equal(symbol)
+
+    const provider = waffle.provider
+    const network = await provider.getNetwork()
+    const deployedToChainId = network.chainId
+    expect(deployedToChainId).to.equal(chainId)
+  })
+
+  it('should match off-chain hash to on-chain hash', async () => {
```

Comment:
Hmm.. negative case coverage is missing I think. Could we add?

Reply:
can you elaborate on this one? I don't follow

Reply:
actually, scratch that. My bad. It won't make sense to have a negative case coverage here.